lib/phoenix/pubsub/pg2_server.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Phoenix.PubSub.PG2Server do
+  use GenServer
+
+  @moduledoc """
+  `Phoenix.PubSub` adapter based on `:pg2`
+
+  See `Phoenix.PubSub.Redis` for details and configuration options.
+  """
+
+  def start_link(opts) do
+    GenServer.start_link __MODULE__, opts, name: Dict.fetch!(opts, :name)
+  end
+
+  def init(opts) do
+    server_name   = Keyword.fetch!(opts, :name)
+    local_name    = Keyword.fetch!(opts, :local_name)
+    pg2_namespace = pg2_namespace(server_name)
+
+    Process.flag(:trap_exit, true)
+
+    :ok = :pg2.create(pg2_namespace)
+    :ok = :pg2.join(pg2_namespace, self)
+
+    {:ok, %{local_name: local_name, namespace: pg2_namespace}}
+  end
+
+  def handle_call({:subscribe, pid, topic, link}, _from, state) do
+    if link, do: Process.link(pid)
```

Comment:
I think the link should be in the local one. After all, the local one is the responsible for tracking local stuff. If we do so, we can remove the trap exit from this process.